doc/src/modules/printing.rst

```diff
@@ -307,14 +307,29 @@ This class implements Python printing. Usage::
     x = Symbol('x')
     e = 5*x**3 + sin(x)
 
-ReprPrinter
------------
+Srepr
+-----
 
 .. module:: sympy.printing.repr
 
 This printer generates executable code. This code satisfies the identity
 ``eval(srepr(expr)) == expr``.
 
+``Srepr()`` gives more low level textual output than ``repr()``
+
+Example::
+
+  >>> repr(5*x**3 + sin(x))
+  '5*x**3 + sin(x)'
+
+  >>> srepr(5*x**3 + sin(x))
+  "Add(Mul(Integer(5), Pow(Symbol('x'), Integer(3))), sin(Symbol('x')))"
+
+``Srepr()`` gives the ``repr`` form, which is what reprwould normally give
+but for SymPy we don’t actually use ``Srepr()`` for ``__repr__`` because it’s 
+is so verbose, it is unlikely that anyone would want it called by default.
```

Comment:
The other reason is that lists call repr on their elements, like `print([a, b, c])` calls `repr(a)`, `repr(b)`, `repr(c)`. So if we used `srepr` for `__repr__` any list with SymPy objects would include the srepr form, even if you used `str` or `print`.